platform/viewer/src/connectedComponents/ToolbarRow.js

```diff
@@ -88,13 +88,20 @@ class ToolbarRow extends Component {
   }
 
   componentDidUpdate(prevProps) {
+    // TODO: This does not compare the arrays correctly
```

Comment:
Not sure if I get it, maybe we should add why it's not correct or maybe raise an issue to fix this? what do you think?

Reply:
Arrays can not be compared just by doing: `ArrayA <conditional> ArrayB`
This was the root cause I've added the comment. 

I rechecked the issue, debugged for a bit and figured out the issue why it was not working properly if I use the correct way of campare arrays (I've created a function in this PR to do it). 

This PR is dependent on anoter PR now #1360 
We need to get this (#1360) merged first, and then merge this one because E2E tests will break.